overrideAttrs only works if the attribute is already populated

if you try 

```
someProgram.overrideAttrs(old: {
    configureFlags = old.configureFlags ++ ["--mimic-threaten-tag"];
})
```
configureFlags has to be populated, if it's empty overriderAttrs can't append to an empty list

```
nix repl
:l <nixpkgs>

nix-repl> cowsay.configureFlags
[ ]

cowsay.overrideAttrs(old: { configureFlags = old.configureFlags ++  ["--dummy"]; })


error: attribute 'configureFlags' missing

       at «string»:1:45:

            1| cow.overrideAttrs(old: { configureFlags = old.configureFlags ++  ["--dummy"]; })
             |                                             ^
            2|

```

however

```
nix-repl> cowsay.outputs                                                           
[ "out" "man" ]

cowsay.overrideAttrs(old: { outputs = old.outputs ++  ["--dummy"]; })
«derivation /nix/store/0fva8szi6s2cz838l9ag9xayd583mxay-cowsay-3.04.drv»
```

so i think this is a quirk with overrideAttrs

however is easily fixed with

```
:b  hello.overrideAttrs(old: { configureFlags = ["--dummy"]; }) 

error: builder for '/nix/store/k5y89y29rgkrwapvak19giwbhkpncabv-hello-2.10.drv' failed with exit code 1;
       last 9 log lines:
       > unpacking sources
       > unpacking source archive /nix/store/3x7dwzq014bblazs7kq20p9hyzz0qh8g-hello-2.10.tar.gz
       > source root is hello-2.10
       > setting SOURCE_DATE_EPOCH to timestamp 1416139241 of file hello-2.10/ChangeLog
       > patching sources
       > configuring
       > configure flags: --disable-dependency-tracking --prefix=/nix/store/133si7nj0lpg8bf0hwgpi7wbq8nmzbla-hello-2.10 --dummyaaa
       > configure: error: unrecognized option: `--dummyaaa'
       > Try `./configure --help' for more information
       For full logs, run 'nix log /nix/store/k5y89y29rgkrwapvak19giwbhkpncabv-hello-2.10.drv'.
```


bizzare huh! i presume the there is no 'old.configureFlags', because it's empty :-) .... quirky?

--- a/README.md
+++ b/README.md
@@ -522,13 +522,20 @@ like this:
 
 ```nix
 someProgram.overrideAttrs(old: {
-    configureFlags = old.configureFlags ++ ["--mimic-threaten-tag"];
+    configureFlags = old.configureFlags or [] ++ ["--mimic-threaten-tag"];
 })
 ```
 
 This pattern has a variety of applications of varying complexity. The top-level
 package set itself can have an `overlays` argument passed to it which may add
 new packages to the imported set.
+
+Note the use of the `or` operator to default to an empty list if the
+original flags do not include `configureFlags`. This is required in
+case a package does not set any flags by itself.
+
+Since this can change in a package over time, it is useful to guard
+against it using `or`.
 
 For a slightly more advanced example, assume that we want to import `<nixpkgs>`
 but have the modification above be reflected in the imported package set:
@@ -537,7 +544,7 @@ but have the modification above be reflected in the imported package set:
 let
   overlay = (self: super: {
     someProgram = super.someProgram.overrideAttrs(old: {
-      configureFlags = old.configureFlags ++ ["--mimic-threaten-tag"];
+      configureFlags = old.configureFlags or [] ++ ["--mimic-threaten-tag"];
     });
   });
 in import <nixpkgs> { overlays = [ overlay ]; }

